### PR TITLE
New Package: ncspot-0.2.0.

### PR DESCRIPTION
**Pull request by ThatGeekyWeeb**

### CODE REVIEW

File: srcpkgs/ncspot/template [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# Template file for 'ncspot'
+pkgname=ncspot
+version=0.2.0
+revision=1
+wrksrc="ncspot-0.2.0"
+build_style="cargo"
+hostmakedepends="cargo"
```
Comment (Chocimier) [inline srcpkgs/ncspot/template:7]: provided by build style, remove

File: srcpkgs/ncspot/template [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+makedepends="ncurses-devel dbus-devel xcb-util-renderutil-devel libxcb-devel libressl-devel pulseaudio-devel"
+short_desc="Spotify client written in Rust using librespot and ncurses"
+maintainer="ThatGeekyWeeb <thatgeekyweeb@gmail.com>"
+license="BSD-2-Clause"
+homepage="https://github.com/hrkfdn/ncspot"
+distfiles="https://github.com/hrkfdn/ncspot/archive/v${version}.tar.gz"
+checksum="876553020e53b8bbf7dac1ddf18f69115d622b302506ce5dc1c3bac1b59ef39b"
+pre_build() {
+	vlicense ${FILESDIR}/LICENSE "BSD-2-Clause"
```
Comment (Chocimier) [inline srcpkgs/ncspot/template:16]: vlicense usualy goes to post_install

File: srcpkgs/ncspot/template [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+	cd ${wrksrc}
```
Comment (Chocimier) [inline srcpkgs/ncspot/template:17]: cd at end of function is no-op

**Comment by ThatGeekyWeeb** [comment]
Builds on arm based aches fail due to
> `thread 'main' panicked at '/builddir/ncspot-0.2.0/target/aarch64-unknown-linux-gnu/release/build/ncurses-00645f4d6448e2e2/out/chtype_size failed: Os { code: 8, kind: Other, message: "Exec format error" }', /host/cargo/registry/src/github.com-1ecc6299db9ec823/ncurses-5.99.0/build.rs:106:20`

I cannot confirm the cause of this, nor can I figure out how to fix it,
even cloning `makedepends` and `hostmakedepends` did not resolve the issue, thus I am restricting the arches
> `makedepends="${hostmakedepends}"`

**Comment by fosslinux** [comment]
No, the issue here is not restricted archs, it is that the package is nocross (ie, will need some fiddling with `qemu-user-static` and stuff to generate the offending data). Please change the `archs=` line to `nocross="ncurses exec format error"` or something of that nature.

**Comment by fosslinux** [comment]
Also, conflicts with #23415

**Comment by ThatGeekyWeeb** [comment]
Understandable, although, #23415 seems to have a much nicer template, I'm a bit new so I figured it would be, I don't know if I'll close this PR or not yet, but if #23415 got updated for `"0.2.0"` I most likely would, seeing as "`0.2.0`" is the stable release for everything other than `nixpkgs` and `DPorts`

**Comment by ThatGeekyWeeb** [comment]
For now I'll just make the requested edits

File: srcpkgs/ncspot/template [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+revision=1
+build_style="cargo"
+makedepends="ncurses-devel python3 dbus-devel xcb-util-renderutil-devel libxcb-devel libressl-devel pulseaudio-devel pkg-config"
+short_desc="Spotify client written in Rust using librespot and ncurses"
+maintainer="ThatGeekyWeeb <thatgeekyweeb@gmail.com>"
+license="BSD-2-Clause"
+homepage="https://github.com/hrkfdn/ncspot"
+distfiles="https://github.com/hrkfdn/ncspot/archive/v${version}.tar.gz"
+checksum="876553020e53b8bbf7dac1ddf18f69115d622b302506ce5dc1c3bac1b59ef39b"
+nocross=yes#Reason: "ncurses exec format error"
```
Comment (ericonr) [inline srcpkgs/ncspot/template:13]: `nocross="ncurses exec format error"`

File: srcpkgs/ncspot/template [hunk 5]
```diff
@@ -0,0 +1,16 @@
+# Template file for 'ncspot'
+pkgname=ncspot
+version=0.2.0
+revision=1
+build_style="cargo"
+makedepends="ncurses-devel python3 dbus-devel xcb-util-renderutil-devel libxcb-devel libressl-devel pulseaudio-devel pkg-config"
```
Comment (ericonr) [inline srcpkgs/ncspot/template:6]: `pkg-config` should be a `hostmakedepends`

Try to keep your lines to a limit of 80 characters too.

**Comment by rien** [comment]
Hi there! ncspot has different audio backends (alsa, pulse, ...) that's why my PR has the `vopt_if` statements in them (just like the `spotifyd` template, which is very similar).

I can update my PR if you'd like. It was ready for review but I guess it disappeared into he backlog :)

**Comment by ThatGeekyWeeb** [comment]
Hey! Yeh, you template is quite nice, I tend to write my template from the ground up, and leave optional features up to the user to figure out, If you were to update it to `0.2.0` that wouldn't be a bad Idea, sad that it got lost and never got merged, I'd recommend you, re-request review and see what happens from there, if and when you update your PR to `0.2.0` I'll close this PR

**Comment by ThatGeekyWeeb** [comment]
I'm going close this PR in hopes that #23415 will be merged.

**Pull request closed without merging**